pkg/resources/kafka/configs.go

```diff
@@ -41,26 +44,30 @@ func (r *Reconciler) reconcilePerBrokerDynamicConfig(brokerId int32, brokerConfi
 		}
 	}()
 
-	fullPerBrokerConfig := util.ParsePropertiesFormat(brokerConfig.Config)
+	fullPerBrokerConfig, err := properties.NewFromString(brokerConfig.Config)
+	if err != nil {
+		return errors.WrapIf(err, "could not parse broker configuration")
+	}
 
 	currentPerBrokerConfigState := r.KafkaCluster.Status.BrokersState[strconv.Itoa(int(brokerId))].PerBrokerConfigurationState
-	if len(fullPerBrokerConfig) == 0 && currentPerBrokerConfigState != v1beta1.PerBrokerConfigOutOfSync {
+	if fullPerBrokerConfig.Len() == 0 && currentPerBrokerConfigState != v1beta1.PerBrokerConfigOutOfSync {
 		return nil
 	}
 
 	// overwrite configs from configmap
-	configsFromConfigMap := util.ParsePropertiesFormat(configMap.Data[kafka.ConfigPropertyName])
+	configsFromConfigMap, err := properties.NewFromString(configMap.Data[kafka.ConfigPropertyName])
+	if err != nil {
+		return errors.WrapIf(err, "could not parse broker configuration from configmap")
+	}
 	for _, perBrokerConfig := range kafka.PerBrokerConfigs {
-		if configValue, ok := configsFromConfigMap[perBrokerConfig]; ok {
-			fullPerBrokerConfig[perBrokerConfig] = configValue
+		if configProperty, ok := configsFromConfigMap.Get(perBrokerConfig); ok {
+			fullPerBrokerConfig.Put(configProperty)
 		}
 	}
 
-	// query the current configs
-	brokerConfigKeys := make([]string, 0, len(fullPerBrokerConfig))
-	for key := range fullPerBrokerConfig {
-		brokerConfigKeys = append(brokerConfigKeys, key)
-	}
+	// query the current config
+	brokerConfigKeys := fullPerBrokerConfig.Keys()
+	sort.Strings(brokerConfigKeys)
```

Comment:
Why is the sort necessary?

Reply:
Nope. You are right. Sorting is not needed. Will remove it.